modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultGenerator.java

```diff
@@ -343,6 +344,12 @@ private void configureOpenAPIInfo() {
             config.additionalProperties().put("unescapedAppDescription", info.getDescription());
         }
 
+        if (this.openAPI.getSpecVersion().equals(SpecVersion.V31) && !StringUtils.isEmpty(info.getSummary())) {
```

Comment:
The swagger lib yields `null` for `info.getSummary()` in non-3.1 specs, so this guard could potentially be removed, however as we don't provide a fallback like we do for the `description` property, I find the spec version guard more obvious for any future readers of this code.